File: src/main/java/ordilov/randomplay/playlist/domain/youtube/Thumbnails.java
package ordilov.randomplay.playlist.domain.youtube;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;

@Getter
public class Thumbnails {

  @JsonProperty("default")
  private Thumbnail defaultThumbnail;

  @JsonProperty("medium")
  private Thumbnail mediumThumbnail;

  @JsonProperty("high")
  private Thumbnail highThumbnail;

  @JsonProperty("standard")
  private Thumbnail standardThumbnail;

  @JsonProperty("maxres")
  private Thumbnail maxresThumbnail;

  @Getter
  private static class Thumbnail {

    String url;
    int width;
    int height;
  }
}

File: src/main/java/ordilov/lissn/playlist/domain/youtube/Thumbnails.kt
package ordilov.lissn.playlist.domain.youtube

import com.fasterxml.jackson.annotation.JsonProperty

data class Thumbnails(
    @JsonProperty("default")
    val defaultThumbnail: Thumbnail,

    @JsonProperty("medium")
    val mediumThumbnail: Thumbnail,

    @JsonProperty("high")
    val highThumbnail: Thumbnail,

    @JsonProperty("standard")
    val standardThumbnail: Thumbnail,

    @JsonProperty("maxres")
    val maxresThumbnail: Thumbnail
)

data class Thumbnail(
    val url: String,
    val width: Int,
    val height: Int
)